sympy/core/function.py

```diff
@@ -1235,7 +1235,7 @@ def __new__(cls, expr, *variables, **assumptions):
 
             if unhandled_non_symbol:
                 obj = None
-            elif not count.is_Integer:
+            elif not count.is_Integer and not count.is_integer:
```

Comment:
I'm not sure about this edit either. This should be the case where `count` is symbolic.